divi/src/masternode.cpp

```diff
@@ -276,9 +276,10 @@ bool CMasternode::UpdateFromNewBroadcast(CMasternodeBroadcast& mnb)
         addr = mnb.addr;
         lastTimeChecked = 0;
         int nDoS = 0;
-        if (mnb.lastPing == CMasternodePing() || (mnb.lastPing != CMasternodePing() && mnb.lastPing.CheckAndUpdate(nDoS, false))) {
+        if (mnb.lastPing != CMasternodePing() && mnb.lastPing.CheckAndUpdate(*this, nDoS, false)) {
             lastPing = mnb.lastPing;
```

Comment:
`lastPing = mnb.lastPing`
Can be removed since it's already being changed in passing *this to the CheckAndUpdate method, no?

Reply:
Yes indeed, good catch.